autograder/question.py

```diff
@@ -184,6 +204,9 @@ def __init__(self, name = '', max_points = 0,
         self.score = score
         self.message = message
 
+        self.hard_fail = False
```

Comment:
I would also pass this in as an arg and default to False.
This will allow from_dict() to work cleanly and allow people to easily make a hard failing result (e.g., in a test).